x/staking/src/keeper/mod.rs

```diff
@@ -504,11 +496,11 @@ impl<
         // need to be transferred to the NotBonded pool.
         // Compare and subtract the respective amounts to only perform one transfer.
         // This is done in order to avoid doing multiple updates inside each iterator/loop.
-        match amt_from_bonded_to_not_bonded.cmp(&amt_from_not_bonded_to_bonded) {
+        match amt_from_not_bonded_to_bonded.cmp(&amt_from_bonded_to_not_bonded) {
```

Comment:
Please, check this change, `master` has valid logic for it.

Reply:
After discussing offline we agree that this code is in fact correct.